cmake/configs/base.cmake

```diff
@@ -66,6 +66,13 @@ config_bool(
     DEFAULT ON
 )
 
+config_bool(
+    WITH_PIC
```

Comment:
This new option feels like an enhancement rather than feature parity with Autoconf? I guess the existing Autoconf code won't apply PIC for static builds.

Reply:
That's kind of true, our autoconf build doesn't expose an explicit option to control PIC settings, making the introduction of this option an enhancement. 
Though Autoconf is a bit smarter in this regard where this option wouldn't have been necessary in the old build system since it uses Libtool. Autoconf/Libtool can determine when to generate position independent sources without the user needing to explicitly pass `fPIC`. When necessary, it creates position independent objects for target libraries declared with the `la` tag i.e. `libwiredtiger_la` (Libtool essentially exists for this reason, being a platform portable solution for generating position independent code, before modern linking tools properly caught up).
CMake is a bit more simple in this regard and won't generate a static library with PIC unless you explicitly tell it to. I wanted to maintain this ability (allowing us to use PIC on static builds, matching what we do with`libtool`), with the added enhancement of being able to turn it on/off, since PIC isn't always necessary for a static build i.e. if your just linking wiredtiger to an executable you don't need PIC.

Reply:
Thanks for the detailed explanation. I like the flexibility that comes with this "enhancement" :)